New Hampton Bay fan controller #1357

dwest576 opened the pull request.

Koenkk commented on devices.js:
```diff
@@ -138,6 +138,42 @@ const execute = (device, actions, callback, delay) => {
+            fz.generic_state, fz.light_brightness, fz.ignore_fan_change, fz.generic_fan_mode, fz.ignore_onoff_change,
+            fz.ignore_level_report,
+        ],
+        toZigbee: [tz.on_off, tz.light_onoff_brightness, tz.light_brightness, tz.fan_mode],
```
I think these should be removed: `tz.light_onoff_brightness, tz.light_brightness`.

dwest576 replied:
I am not sure if tz.light_onoff_brightness is needed, but tz.light_brightness should remain as the fan does have a light attached and the brightness command does function.

Koenkk replied:
Ok, I wasn't aware of the light, about the on/off converter, is it used to turn on/off the fan or the light? (meaning that the `fan_mode` converter is only used to control the fan).

dwest576 replied:
I believe that tz.on_off controls just the light.  Assuming when I pass "state: off" to qitt that is sends a tz.on_off command.  When I have controlled the fan it has always been through fan_mode.  There is a button on the remote that comes with the controller itself, that is a "all" power button, which turns off the light and the fan all at once.  I have never used it, but it is there.

Koenkk commented:
Are you using Home Assistant?

dwest576 commented:
> Are you using Home Assistant?

Yes, just haven't gotten to that part of the integration yet.   Also, not running Z2M on the hass devices.

Koenkk commented on devices.js:
```diff
@@ -138,6 +138,42 @@ const execute = (device, actions, callback, delay) => {
+        description: 'Universal wink enabled white ceiling fan premier remote control',
+        supports: 'on/off',
+        fromZigbee: [
+            fz.generic_state, fz.light_brightness, fz.ignore_fan_change, fz.generic_fan_mode, fz.ignore_onoff_change,
```
The `generic_fan_mode` converter is missing, can you also add it?

dwest576 replied:
the device insert was from https://github.com/Koenkk/zigbee2mqtt/issues/1357

I'll reach out to the author for that code.

dwest576 commented:
This looks awesome, anything else you need from me.  I am happy to test everything and report back.

Koenkk commented:
@dwest576 could you provide me the `fz.generic_fan_mode`? It's missing from the changed files.

ccalica commented:
Thanks for doing this.  I've been eyeing this controller for awhile.  Just ordered, will test/help when it arrives.

Koenkk commented:
This should be ready to merge now, @dwest576 can you do some final tests to ensure everything still works?

dwest576 commented:
> This should be ready to merge now, @dwest576 can you do some final tests to ensure everything still works?

I'll test it when I get home today, do I need to rebuild or simply place the changed files in the proper folder?

Koenkk commented:
just replace the changed files.

dwest576 commented:
> just replace the changed files.

I tested this evening.   Everything works great.   I paired the device, it pulls everything in, no errors.  I press all the buttons on the remote and I see all the changes in the z2m log.   I sent many commands from mqtt to z2m and it passes them all via zigbee to the fan it all works.

The only thing that doesn't work yet is the auto config into homeasisstant.   I replaced homeassistant.js with the new code and when I pari the fan it doesn't do anything to ha.

Koenkk commented:
I've integrated this in the zigbee2mqtt dev branch, meaning that home assistant discovery should also work there. Please let me know :smile:

dwest576 commented:
At this point, can I just do a new install and pull all this in and test?

> On Apr 26, 2019, at 12:58 PM, Koen Kanters <notifications@github.com> wrote:
> 
> I've integrated this in the zigbee2mqtt dev branch, meaning that home assistant discovery should also work there. Please let me know 😄
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Koenkk/zigbee-shepherd-converters/pull/416#issuecomment-487182258>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG6NABD4M2NRDPU5VN25WIDPSNNFFANCNFSM4HHMOHJQ>.
>

Koenkk commented:
See http://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html

dwest576 commented:
I was able to switch to the dev branch and the fan now gets pushed to homeassistant.   Would it be possible to get the light pushed to homeassistant too?


I did test all the options from homeassistant to the fan and they all work great!

On April 26, 2019 at 11:12 PM, Koen Kanters <notifications@github.com> wrote:



See http://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

Koenkk commented:
Forgot to add that, added it know :smile:

dwest576 commented:
Just updated Dev, and it is now pushing light to Home Assistant.   I’ll test this evening and make sure it works as intended.   Thanks for all your help!

> On Apr 28, 2019, at 11:54 AM, Koen Kanters <notifications@github.com> wrote:
> 
> Forgot to add that, added it know 😄
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/Koenkk/zigbee-shepherd-converters/pull/416#issuecomment-487405897>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AG6NABCTEK3TBPXB5ZNWC23PSXXEVANCNFSM4HHMOHJQ>.
>
